src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs

```diff
@@ -0,0 +1,1118 @@
+        public static Vector128<byte> DecryptLast(Vector128<byte> value, Vector128<byte> roundKey) { throw null; }
+        public static Vector128<sbyte> Encrypt(Vector128<sbyte> value, Vector128<sbyte> roundKey) { throw null; }
+        public static Vector128<byte> Encrypt(Vector128<byte> value, Vector128<byte> roundKey) { throw null; }
+        public static Vector128<sbyte> EncryptLast(Vector128<sbyte> value, Vector128<sbyte> roundKey) { throw null; }
+        public static Vector128<byte> EncryptLast(Vector128<byte> value, Vector128<byte> roundKey) { throw null; }
+        public static Vector128<sbyte> InvisibleMixColumn(Vector128<sbyte> value) { throw null; }
+        public static Vector128<byte> InvisibleMixColumn(Vector128<byte> value) { throw null; }
+        public static Vector128<sbyte> KeygenAssist(Vector128<sbyte> value, byte control) { throw null; }
+        public static Vector128<byte> KeygenAssist(Vector128<byte> value, byte control) { throw null; }
+    }
+    public static class Avx
+    {
+        public static bool IsSupported { get { throw null; } }
+        public static Vector256<float> Add(Vector256<float> left, Vector256<float> right) { throw null; }
+        public static Vector256<double> Add(Vector256<double> left, Vector256<double> right) { throw null; }
+        public static Vector256<float> AddSubtract(Vector256<float> left, Vector256<float> right) { throw null; }
+        public static Vector256<double> AddSubtract(Vector256<double> left, Vector256<double> right) { throw null; }
+        public static Vector256<float> And(Vector256<float> left, Vector256<float> right) { throw null; }
+        public static Vector256<double> And(Vector256<double> left, Vector256<double> right) { throw null; }
+        public static Vector256<float> AndNot(Vector256<float> left, Vector256<float> right) { throw null; }
+        public static Vector256<double> AndNot(Vector256<double> left, Vector256<double> right) { throw null; }
+        public static Vector256<float> Blend(Vector256<float> left, Vector256<float> right, byte control) { throw null; }
+        public static Vector256<double> Blend(Vector256<double> left, Vector256<double> right, byte control) { throw null; }
+        public static Vector256<float> BlendVariable(Vector256<float> left, Vector256<float> right, Vector256<float> mask) { throw null; }
+        public static Vector256<double> BlendVariable(Vector256<double> left, Vector256<double> right, Vector256<double> mask) { throw null; }
+        public static Vector128<float> BroadcastElementToVector128(ref float source) { throw null; }
+        public static unsafe Vector256<float> BroadcastVector128ToVector256(float* address) { throw null; }
+        public static Vector256<float> BroadcastElementToVector256(ref float source) { throw null; }
+        public static unsafe Vector256<double> BroadcastVector128ToVector256(double* address) { throw null; }
```

Comment:
💭 I'm not sure we need this signature. Specifically, I would prefer either of the following provided we can do it without a negative performance impact:

Option 1: still loading from a memory location:

```csharp
public static Vector256<double> BroadcastVector128ToVector256(ref Vector128<double> source);
```

Option 2: Separate the loading from memory from the broadcast:

```csharp
public static Vector256<double> BroadcastVector128ToVector256(Vector128<double> source);

// Used with Load:
Avx.BroadcastVector128ToVector256(Sse2.Load(location))
```

Reply:
This function `BroadcastVector128ToVector256` will be compiled to AVX instruction `vbroadcastf128` that only has reg+mem operand encoding, and this instruction's semantic is "load and broadcast", so we should not change to Load+Broadcast.